src/nvim/ops.c

```diff
@@ -1672,17 +1672,6 @@ int op_delete(oparg_T *oap)
       extmark_col_adjust_delete(curbuf, lnum, mincol, endcol,
                                 kExtmarkUndo, 0);
     }
-
-  // Delete characters within one line,
-  // The case with multiple lines is handled by do_join
-  } else if (oap->motion_type == kMTCharWise && oap->line_count == 1) {
```

Comment:
maybe assert?

    assert(!(oap->motion_type == kMTCharWise && oap->line_count == 1));  // #11612

Reply:
```
[----------] Running tests from test/functional/api/buffer_updates_spec.lua
[ RUN      ] API: buffer events: when lines are added: 8.31 ms OK
[ RUN      ] API: buffer events: when lines are removed: 6.97 ms OK
[ RUN      ] API: buffer events: when text is changed: -- Tests exited non-zero: SIGPIPE
-- Output to stderr:
nvim: ../src/nvim/ops.c:1677: op_delete: Assertion `!(oap->motion_type == kMTCharWise && oap->line_count == 1)' failed.
```

Reply:
Ah, it's not dead code but the assignments are never used. Will just assume that this is a harmless oversight from the extmarks work...